api/src/test/java/io/opentelemetry/common/AttributesTest.java

```diff
@@ -149,4 +149,41 @@ public void builder_arrayTypes() {
                 "double", arrayAttributeValue(33.44, -44.33),
                 "boolean", arrayAttributeValue(false, true)));
   }
+
+  @Test
+  public void get_Null() {
+    assertThat(Attributes.empty().get("foo")).isNull();
+    assertThat(Attributes.of("key", stringAttributeValue("value")).get("foo")).isNull();
+  }
+
+  @Test
+  public void get() {
+    assertThat(Attributes.of("key", stringAttributeValue("value")).get("key"))
```

Comment:
For `Attributes`, having `getString` etc would make these sort of unit tests (thinking user code unit tests less worried about these speifically) nicer.

Reply:
Not a bad idea. What behavior would you expect if you were asking for a non-string attribute value?

Reply:
I've seen similar APIs return `null` when it's a different type, it seems fine to me.

Reply:
Right now, the corresponding methods on AttributeValue throw an UnsupportedOperationException. I honestly don't know what's the better approach.

Reply:
Observe this is not an API intended to be used by end users, but by 'us', so I'm not so sure the sugar is truly needed.

In any case, if this is something still required, I'd suggest filling an issue and discuss the change there, so we can go ahead and merge this PR soon ;)

Reply:
fair enough, @carlosalberto . I don't think it's needed right now, although it could make exporters easier to write, possibly.

Reply:
Yeah, we shouldn't do this now; and only add it if it is really needed. Resolving this conversation for now.

Reply:
Ok with addressing later. Don't quite understand this being not an end-user API since it would be used to create Resource too?